src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -326,5 +327,96 @@ DiagnosticDataProvider & GetDiagnosticDataProviderImpl()
     return DiagnosticDataProviderImpl::GetDefaultInstance();
 }
 
+#ifdef CONFIG_CHIP_WIFI
```

Comment:
`Kconfig.features` is not mandatory file for users' applications, so as long as `CONFIG_CHIP_WIFI` is defined there, it shouldn't be referred to be the code. `Kconfig.features` is only meant to aggregate groups of settings needed to enable optional functions. Also, we add nrfconnect-only implementation to `Zephyr` platform directory. Maybe we should extract WiFi part from this interface (TODO) or inherit from Zephyr's DiagnosticDataProvider and put nRF Connect-only code in nRF Connect implementation.

Reply:
Done, thanks for the hints.